WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -43,9 +43,35 @@ public class CoinJoinClient
 		public KeyManager Keymanager { get; }
 		private RoundStateUpdater RoundStatusUpdater { get; }
 
-		public async Task StartCoinJoinAsync(CancellationToken cancellationToken, IEnumerable<Money>? forcedOutputDenominations = null)
+		public async Task<bool> StartCoinJoinAsync(CancellationToken cancellationToken, IEnumerable<Money>? forcedOutputDenominations = null)
+		{
+			var currentRoundState = await RoundStatusUpdater.CreateRoundAwaiter(roundState => roundState.Phase == Phase.InputRegistration, cancellationToken).ConfigureAwait(false);
+
+			// This should be roughly log(#inputs), it could be set slightly
+			// higher if more inputs are observed but that involves trusting the
+			// coordinator with those values. Therefore, conservatively set this
+			// so that a maximum of 5 blame rounds are executed.
+			// FIXME should smaller rounds abort earlier?
+			var tryLimit = 6;
+
+			for (var tries = 0; tries < tryLimit; tries++)
+			{
+				if (await StartRoundAsync(currentRoundState, cancellationToken, forcedOutputDenominations))
+				{
+					return true;
+				}
+				else
+				{
+					var blameRoundState = await RoundStatusUpdater.CreateRoundAwaiter(roundState => roundState.BlameOf == currentRoundState.Id, cancellationToken).ConfigureAwait(false);
+					currentRoundState = blameRoundState;
+				}
+			}
+
+			return false;
+		}
+
+		public async Task<bool> StartRoundAsync(RoundState roundState, CancellationToken cancellationToken, IEnumerable<Money>? forcedOutputDenominations = null)
```

Comment:
In this commit I remove the optional parameter https://github.com/zkSNACKs/WalletWasabi/pull/5827/commits/0bcbd049add80ad737d19d7743217daf1d259c03 Basically just pass what you want the method to use.

Reply:
the only dependency on that parameter is from the tests, and it was only added to debug something that is no longer relevant, it has no use in the real code so I think we should just remove it